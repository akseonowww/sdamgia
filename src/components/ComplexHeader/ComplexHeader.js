import React from 'react';

import './ComplexHeader.scss';

const ComplexHeader = props => {
	const { className, children } = props;

	return <div className={`ComplexHeader ${className}`}>{children}</div>;
};

export default ComplexHeader;
