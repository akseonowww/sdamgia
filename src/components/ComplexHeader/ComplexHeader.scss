@import '../../breakpoints';

.ComplexHeader {
	&-Link {
		margin-top: 0.4rem;

		&_mobile {
			margin-top: 1rem;

			text-align: center;

			@include media-breakpoint-up(md) {
				display: none;
			}
		}

		&_desktop {
			display: none;

			@include media-breakpoint-up(md) {
				display: block;
			}
		}
	}
}
