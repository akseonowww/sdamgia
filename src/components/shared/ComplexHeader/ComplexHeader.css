.ComplexHeader-Link {
	margin-top: 5px;
}
