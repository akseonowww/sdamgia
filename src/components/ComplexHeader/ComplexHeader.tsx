import React, { FC, ReactNode } from 'react'
import cx from 'classnames'

import './ComplexHeader.scss'

interface IComplexHeaderProps {
  className?: string
  children: ReactNode
}

const ComplexHeader: FC<IComplexHeaderProps> = ({ className, children }) => (
  <div className={cx('ComplexHeader', className)}>{children}</div>
)

export default ComplexHeader
