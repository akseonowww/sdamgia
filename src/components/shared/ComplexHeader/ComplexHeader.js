import React from 'react';

import './ComplexHeader.css';

const ComplexHeader = props => {
	const { className, children } = props;

	return <div className={'ComplexHeader' + ' ' + className}>{children}</div>;
};

export default ComplexHeader;
