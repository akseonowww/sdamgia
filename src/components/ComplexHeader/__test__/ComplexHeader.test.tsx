import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import ComplexHeader from '../ComplexHeader'

describe('ComplexHeader', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(
        <ComplexHeader>
          <h1>Variants</h1>
          <div>Archive</div>
        </ComplexHeader>,
        container
      )
    })
    expect(container && container.textContent).toBe('VariantsArchive')

    act(() => {
      render(
        <ComplexHeader className="Section-ComplexHeader">
          <h1>Variants</h1>
          <div>Archive</div>
        </ComplexHeader>,
        container
      )
    })
    expect(container && container.textContent).toBe('VariantsArchive')
  })

  it('should render a content', () => {
    act(() => {
      render(
        <ComplexHeader>
          <h1>Variants</h1>
          <div>Archive</div>
        </ComplexHeader>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <ComplexHeader className="Section-ComplexHeader">
          <h1>Variants</h1>
          <div>Archive</div>
        </ComplexHeader>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
