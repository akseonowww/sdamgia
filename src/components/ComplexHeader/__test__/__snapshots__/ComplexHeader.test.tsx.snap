// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplexHeader should render a content 1`] = `
"<div class=\\"ComplexHeader\\">
  <h1>Variants</h1>
  <div>Archive</div>
</div>"
`;

exports[`ComplexHeader should render a content 2`] = `
"<div class=\\"ComplexHeader Section-ComplexHeader\\">
  <h1>Variants</h1>
  <div>Archive</div>
</div>"
`;
